library/src/interactor_impl.cxx

```diff
@@ -558,16 +498,97 @@ class interactor_impl::internals
     this->Window.render();
   }
 
+  struct tokenize_exception : public exception
+  {
+    explicit tokenize_exception(const std::string& what = "")
+      : exception(what){}
+  };
+
+  static std::vector<std::string> Tokenize(std::string_view str)
+  {
+    std::vector<std::string> tokens;
+    std::string token;
+    const auto accumulate = [&](const char &c) { token.push_back(c); };
+    const auto emit = [&]() {
+      if (!token.empty()) {
+        std::cout<<token<<std::endl;
+        tokens.push_back(token);
+        token.clear();
+      }
+    };
+    bool escaped = false;
+    char quoted = '\0';
+    for (const char& c : str)
+    {
+      switch (c)
+      {
+        case '\\':
+          escaped = true;
+          break;
+        case ' ':
+          if (!escaped && !quoted)
+          {
+            emit();
+          }
+          else
+          {
+            accumulate(c);
+          }
+          escaped = false;
+          break;
+        case '"':
+        case '\'':
+        case '`':
+          if (!escaped && quoted == c)
+          {
+            emit();
+            quoted = '\0';
+          }
+          else if (!escaped && !quoted)
+          {
+            quoted = c;
+          }
+          else
+          {
+            accumulate(c);
+          }
+          escaped = false;
+          break;
+        default:
+          if (escaped)
+          {
+            accumulate('\\');
+          }
+          accumulate(c);
+          escaped = false;
+          break;
+      }
+    }
+    if (quoted)
+    {
+      throw tokenize_exception();
+    }
+    if (escaped)
+    {
+      accumulate('\\');
+    }
+    emit();
+    return tokens;
+  }
+
   options& Options;
   window_impl& Window;
   scene_impl& Scene;
+  interactor_impl& Interactor;
   animationManager* AnimationManager;
 
   vtkSmartPointer<vtkRenderWindowInteractor> VTKInteractor;
   vtkNew<vtkF3DInteractorStyle> Style;
   vtkSmartPointer<vtkF3DInteractorEventRecorder> Recorder;
   std::map<unsigned long, std::pair<int, std::function<void()>>> TimerCallBacks;
 
+  std::map<std::string, std::function<bool(const std::vector<std::string>&)>> CommandCallbacks;
```

Comment:
I know that `std::unordered_map` needs a hash function but it's faster and we do not need sorting.

Reply:
really does not matter for small maps imo.

Reply:
I agree but in practice we do not know the size of the map. That's not super important for now I guess anyway.

Reply:
the same argument can be made about almost any of the maps in the libf3d. Not worth it imo.